### PR TITLE
Add unit tests and some code clean up fixes gh-#196

### PR DESCRIPTION
**Pull request by nicce**

### CODE REVIEW

File: src/main/java/com/jadventure/game/JAdventure.java [hunk 1]
```diff
@@ -45,8 +45,6 @@ public static void main(String[] args) {
                         Runnable r = new MainMenu(server, mode);
                         new Thread(r).start();
                     }
-                } catch (SocketException e) {
```
Comment (Dev-Osmium) [inline src/main/java/com/jadventure/game/JAdventure.java]: Sorry if I'm not understanding this, but why was this removed?

Comment (Dev-Osmium) [reply src/main/java/com/jadventure/game/JAdventure.java]: Nevermind, I can't think today. Sorry! Looks good to me.

**Comment by Dev-Osmium** [comment]
I think this is a good first start. Full code coverage would be nice, but the more the better.

**Comment by paddatrapper** [comment]
Sorry for the delay in merging this...
